dotCMS/src/main/java/com/dotcms/rest/api/v1/system/monitor/MonitorHelper.java

```diff
@@ -10,81 +10,58 @@
-import java.util.concurrent.Callable;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.Future;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
 
-import static com.dotcms.content.elasticsearch.business.ESIndexAPI.INDEX_OPERATIONS_TIMEOUT_IN_MS;
+class MonitorHelper {
 
 
-class MonitorHelper {
-    private static final long   DEFAULT_LOCAL_FS_TIMEOUT    = 1000;
-    private static final long   DEFAULT_CACHE_TIMEOUT       = 1000;
-    private static final long   DEFAULT_ASSET_FS_TIMEOUT    = 1000;
-    private static final long   DEFAULT_INDEX_TIMEOUT       = 1000;
-    private static final long   DEFAULT_DB_TIMEOUT          = 1000;
-    private static final String[] DEFAULT_IP_ACL_VALUE        = new String[] {"127.0.0.1/32","10.0.0.0/8","172.16.0.0/12","192.168.0.0/16"};
-
-
-    private static final String SYSTEM_STATUS_API_IP_ACL           = "SYSTEM_STATUS_API_IP_ACL";
-    private static final String SYSTEM_STATUS_API_LOCAL_FS_TIMEOUT = "SYSTEM_STATUS_API_LOCAL_FS_TIMEOUT";
-    private static final String SYSTEM_STATUS_API_CACHE_TIMEOUT    = "SYSTEM_STATUS_API_CACHE_TIMEOUT";
-    private static final String SYSTEM_STATUS_API_ASSET_FS_TIMEOUT = "SYSTEM_STATUS_API_ASSET_FS_TIMEOUT";
-    private static final String SYSTEM_STATUS_API_INDEX_TIMEOUT    = "SYSTEM_STATUS_API_INDEX_TIMEOUT";
-    private static final String SYSTEM_STATUS_API_DB_TIMEOUT       = "SYSTEM_STATUS_API_DB_TIMEOUT";
-
-    private static final int SYSTEM_STATUS_CACHE_RESPONSE_SECONDS = Config.getIntProperty("SYSTEM_STATUS_CACHE_RESPONSE_SECONDS",10);
-    
-    private static final String[] ACLS_IPS = Config.getStringArrayProperty(SYSTEM_STATUS_API_IP_ACL, DEFAULT_IP_ACL_VALUE);
+    private static final String[] DEFAULT_IP_ACL_VALUE = new String[]{"127.0.0.1/32", "10.0.0.0/8", "172.16.0.0/12",
+            "192.168.0.0/16"};
+
+
+    private static final String SYSTEM_STATUS_API_IP_ACL = "SYSTEM_STATUS_API_IP_ACL";
 
 
-    private static final long localFSTimeout = Config.getLongProperty(SYSTEM_STATUS_API_LOCAL_FS_TIMEOUT, DEFAULT_LOCAL_FS_TIMEOUT);
-    private static final long cacheTimeout  = Config.getLongProperty(SYSTEM_STATUS_API_CACHE_TIMEOUT, DEFAULT_CACHE_TIMEOUT);
-    private static final long assetTimeout = Config.getLongProperty(SYSTEM_STATUS_API_ASSET_FS_TIMEOUT, DEFAULT_ASSET_FS_TIMEOUT);
-    private static final long indexTimeout = Config.getLongProperty(SYSTEM_STATUS_API_INDEX_TIMEOUT, DEFAULT_INDEX_TIMEOUT);
-    private static final long dbTimeout = Config.getLongProperty(SYSTEM_STATUS_API_DB_TIMEOUT, DEFAULT_DB_TIMEOUT);
+    private static final int SYSTEM_STATUS_CACHE_RESPONSE_SECONDS = Config.getIntProperty(
+            "SYSTEM_STATUS_CACHE_RESPONSE_SECONDS", 10);
+
+    private static final String[] ACLS_IPS = Config.getStringArrayProperty(SYSTEM_STATUS_API_IP_ACL,
+            DEFAULT_IP_ACL_VALUE);
 
     
+    static Tuple2<Long, MonitorStats> cachedStats = null;
```

Comment:
This is not thread safe, changes to the value stored in cachedStats may not be visible to other threads without it being set as volatile.  Can be safer to use an AtomicReference to hold the Tuple2<Long, MonitorStats> which also gives methods to be able to atomically update it also.

Reply:
changed to `AtomicReference`